File: CivitasJava/src/civitas/SorpresaIrCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author luis
 */
public class SorpresaIrCarcel extends Sorpresa{
    private Tablero tablero;
    
    SorpresaIrCarcel(Tablero tablero){
        super("SopresaIrCarcel");
        this.tablero = tablero;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos)){
            informe(actual,todos);
            todos.get(actual).encarcelar(tablero.getCarcel());
        }
        
    }
    
    @Override
    public String toString(){
        String e = "\nNombre: SorpresaIrCarcel" ;
        return e;
    }
}

File: CivitasRuby/lib/sorpresa_ir_carcel.rb
# encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class SorpresaIrCarcel < Sorpresa
    
    def initialize(tablero)
      super("SorpresaIrCarcel")
      @tablero = tablero
    end
        
    def aplicar_a_jugador(actual, todos)
      if jugador_correcto(actual,todos)
        informe(actual,todos)
        todos[actual].encarcelar(@tablero.numCasillaCarcel)
      end
    end
    
    def to_s
      "\nNombre: #{@texto}"
    end

       
    public_class_method  :new
    
  end
end
